apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c

```diff
@@ -0,0 +1,244 @@
+
+	TC_SUCCESS_RESULT();
+}
+
+
+static void utc_wifi_manager_connect_ap_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	/* Connect to AP */
+	wifi_manager_ap_config_s config = {\
+		"NETGEAR99", 9,\
+		"artikwifi", 9, \
+		WIFI_MANAGER_AUTH_WPA2_PSK, \
+		WIFI_MANAGER_CRYPTO_AES
+	};
+
+	/* current wifi mode is softap, then this try will fail */
+	ret = wifi_manager_connect_ap(&config);
+
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_connect_ap_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	/* change to station mode */
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+
+	/* Connect to AP */
+	wifi_manager_ap_config_s config = {\
+		"NETGEAR99", 9,\
+		"artikwifi", 9, \
+		WIFI_MANAGER_AUTH_WPA2_PSK, \
+		WIFI_MANAGER_CRYPTO_AES
+	};
+
+	/* current wifi mode is softap, then this try will fail */
+	ret = wifi_manager_connect_ap(&config);
+
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_deinit_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_deinit_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_DEINITIALIZED);
+	TC_SUCCESS_RESULT();
+}
+
+int wifi_manager_utc(int argc, FAR char *argv[])
+{
+	total_pass = 0;
+	total_fail = 0;
+
+	utc_wifi_manager_init_n();
+	utc_wifi_manager_init_p();
+
+	utc_wifi_manager_set_mode_n();
+	utc_wifi_manager_set_mode_p();
+
+	utc_wifi_manager_get_mode_n();
+	utc_wifi_manager_get_mode_p();	// set softap mode inside this function
+
+	/* shoud define ap config first in utc_wifi_manager_connect_ap_n()
+	 * and utc_wifi_manager_connect_ap_p() before trying these two tests
+	 * Otherwise, you will get failed.
+	 */
+	utc_wifi_manager_connect_ap_n();	// try to connect to ap in softap mode
+	utc_wifi_manager_connect_ap_p();	// change to station mode and try to connect to ap
+
+	/* wait for 10 seconds to acquire ip address from the connected ap.
+	 * Even though ip address is not assigned, this unit test may pass if joing ap is successful.
+	 */
+	sleep(10);
+
+	utc_wifi_manager_deinit_p();
+	utc_wifi_manager_deinit_n(); // Should be run after positive tc, that is, the second deinit gets failed.
+
+	printf("\n=== TINYARA WIFI_MANAGER TC COMPLETE ===\n");
+	printf("\t\tTotal pass : %d\n\t\tTotal fail : %d\n", total_pass, total_fail);
+
+	return 0;
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int utc_wifi_manager_main(int argc, char *argv[])
+#endif
+{
+#ifdef CONFIG_TASH
+	int ret = 0;
+	ret = tash_cmd_install("wifi_mgr_utc", wifi_manager_utc, TASH_EXECMD_SYNC);
```

Comment:
This makes tester should type in a command to lauch this. I think we don't need this.

Reply:
this tash command registration is like a step which we have to follow. Please look at other testcases.
In addition, this tash command help us to quickly test testcases.